mantidproject/mantid
# Fix bugs with restoring defaults on the reflectometry GUI

**Report to:** Max at ISIS

This PR fixes a couple of bugs on the ISIS Reflectometry interface to do with settings being overridden:
- The Experiment/Instrument settings were being overridden by defaults if the instrument changed as part of a load.
- The Experiment/Instrument settings on all tabs were being overridden by defaults when you create a new batch. This was due to the instrument being updated when it gets set on the new batch.
- The Experiment/Instrument settings on all batches/tabs were also being overridden when you click Restore Defaults on a single tab. This was due to the instrument being updated before loading defaults, which is fine, but then it was issuing a notification to all batches that an update had happened.

This PR also contains some minor tidying and improvements to unit tests (particularly the decoder) and removes an unused function declaration.

**To test:**

- Open the `ISIS Reflectometry` interface
- From the Batch->Load menu, load the `INTER_NR_test.json` file from the ISIS sample data
- Edit some settings on the Experiment and Instrument tab
- From the Batch->Save menu, save the settings to a new file
- Change the instrument to something other than INTER and re-load the file you saved
- Check that the settings you changed are restored
- Select Batch->New to create a new batch
- Check that the settings you changed in the original batch are not affected
- Click Restore Defaults on an Experiment/Instrument tab and check that only that tab is affected (and no other batches are changed).

Fixes #28429 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
